### ISSUE TITLE
why force .cleanup callback?

### ISSUE THREAD

**Issue by Kikobeats** [ArtskydJ/create-temp-file#11]
I don't understand this piece of code:

```
writeStream.cleanup = function cln(cb) {
        if (!streamHasEnded) writeStream.end()
        fs.unlink(path, cb || emitError)
    }
```

why, if not provide a callback, delete a file in any moment in the future? why is forcing emit an error?

**Comment by ArtskydJ** [comment]
#3 The error was being swallowed.

**Comment by ArtskydJ** [comment]
I didn't want the errors to be swallowed, so I am emitting the error.

**Comment by Kikobeats** [comment]
why not `fs.unlink(path, cb || function() {})` ?

**Comment by ArtskydJ** [comment]
It was so if someone wanted to watch for stream errors or cleanup errors, they could just watch in one place.

**Comment by ArtskydJ** [comment]
You can always ignore the 'error' event...

Previously, if there was no error, this would happen: `ws.emit('error', null)`. This has been fixed in 2dcbe229d7e92759a5363794d9a23176a18fe450.

**Comment by Kikobeats** [comment]
If it is responsibility of the user, not should be added in the library because broken what expect of the library behavior!
